### PR TITLE
dashboard/adapter/manager.go:reduce the complex to O(n) from O(nlogn) for setNewAddr

### LINKED ISSUE #4862: dashboard/manager/setNewAddress:  find the member with min ID directly instead of `sort`
## Enhancement Task

It seems we use the `sort` function to find the minimum item, how about find it directly ? The complex can reduce to O(N) from O(NlogN)

[https://github.com/tikv/pd/blob/c264a8196c970456e7865b50cd2b3a02a0ff86cb/pkg/dashboard/adapter/manager.go#L179-L189](https://github.com/tikv/pd/blob/c264a8196c970456e7865b50cd2b3a02a0ff86cb/pkg/dashboard/adapter/manager.go#L179-L189)

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve?

Issue Number: Close #4862 

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/dashboard/adapter/manager.go [hunk 1]
```diff
@@ -171,25 +170,25 @@ func (m *Manager) needResetAddress(addr string) bool {
...
-		sort.Slice(members, func(i, j int) bool { return members[i].GetMemberId() < members[j].GetMemberId() })
-		for _, member := range members {
-			if member.MemberId != leaderID {
-				addr = member.GetClientUrls()[0]
+	if len(m.members) == 0 {
+		return
+	}
+	// select the sever with minimum member ID(avoid the PD leader if possible) to run dashboard.
+	minMemberIdx := 0
+	if len(m.members) > 0 {
```
Comment (nolouch) [inline pkg/dashboard/adapter/manager.go:178]: can remove this `if`?

**Review by nolouch — COMMENTED** [review]
rest LGTM

File: pkg/dashboard/adapter/manager.go (hunk 1 above)
Comment (AndreMouche) [reply pkg/dashboard/adapter/manager.go:178]: good catch

File: pkg/dashboard/adapter/manager.go [hunk 2]
```diff
@@ -171,25 +170,25 @@ func (m *Manager) needResetAddress(addr string) bool {
...
-	case 1:
-		addr = members[0].GetClientUrls()[0]
-	default:
-		addr = members[0].GetClientUrls()[0]
-		leaderID := m.srv.GetMemberInfo().MemberId
-		sort.Slice(members, func(i, j int) bool { return members[i].GetMemberId() < members[j].GetMemberId() })
-		for _, member := range members {
-			if member.MemberId != leaderID {
-				addr = member.GetClientUrls()[0]
+	if len(m.members) == 0 {
```
Comment (rleungx) [inline pkg/dashboard/adapter/manager.go:173]: It has been guaranteed by the caller.

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**